code/game/objects/structures/popout_cake.dm

```diff
@@ -0,0 +1,185 @@
+/obj/structure/popout_cake
+	name = "towering cake"
+	desc = "A humongous multi-tiered cake."
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "popout_cake"
+	density = TRUE
+	anchored = FALSE
+	can_buckle = 1
+	buckle_lying = 0 //you sit in a chair, not lay
+	resistance_flags = NONE
+	max_integrity = 100
+	integrity_failure = 10
+	move_resist = MOVE_FORCE_WEAK
+	layer = OBJ_LAYER
+	var/mob/living/occupant = null
+	var/datum/action/item_action/pull_string/string //Action for pulling the string for a surprise reveal
+	var/used_string = FALSE //If the string for a surprise reveal has been pulled from inside
+	var/amount_of_slices = 16 //How many cake slices will appear once it's cut up
+	var/slice_path  = /obj/item/reagent_containers/food/snacks/cakeslice/plain/full //What kind of cake slice will appear
+	var/strong_surprise = FALSE //If the surprise reveal has an extra oomph to it, used for the newcop exclusive cake
+
+/obj/structure/popout_cake/newcop
+	strong_surprise = TRUE
+
+/obj/structure/popout_cake/Initialize(mapload)
+	. = ..()
+	if(!string)
+		var/datum/action/item_action/pull_string/S = new()
```

Comment:
this var does not need to exist at all, just initialize directly into `string`

Reply:
Wait, how?

Reply:
Nvm figured it out.